src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1550,6 +1550,27 @@ public ErrorDiagnostic UnknownModuleReferenceScheme(string badScheme, ImmutableA
                 null,
                 DiagnosticStyling.Default,
                 new CodeFix($"Change \"{name}\" to \"{knownFunctionNamespace}.{knownFunctionName}\"", true, CodeFixKind.QuickFix, CodeManipulator.Replace(TextSpan, $"{knownFunctionNamespace}.{knownFunctionName}")));
+
+            public ErrorDiagnostic UnparsableBicepConfigFile(string configurationPath, string parsingErrorMessage) => new(
+                TextSpan,
+                "BCP266",
+                $"Failed to parse the contents of the Bicep configuration file \"{configurationPath}\" as valid JSON: \"{parsingErrorMessage}\".");
+
+            public ErrorDiagnostic UnloadableBicepConfigFile(string configurationPath, string loadErrorMessage) => new(
+                TextSpan,
+                "BCP267",
+                $"Could not load the Bicep configuration file \"{configurationPath}\": \"{loadErrorMessage}\".");
+
+            public ErrorDiagnostic InvalidBicepConfigFile(string configurationPath, string parsingErrorMessage) => new(
+                TextSpan,
+                "BCP268",
+                $"Failed to parse the contents of the Bicep configuration file \"{configurationPath}\": \"{parsingErrorMessage}\".");
+
+            public Diagnostic PotentialConfigDirectoryCouldNotBeScanned(string? directoryPath, string scanErrorMessage) => new(
+                TextSpan,
+                DiagnosticLevel.Info, // should this be a warning instead?
+                "BCP269",
+                $"Error scanning \"{directoryPath}\" for bicep configuration: \"{scanErrorMessage}\".");
```

Comment:
The error-level diagnostics would be thrown as `ConfigurationException`s from the configuration manager. The language server's compilation facade was wrapping calls into the config manager in a try/catch and converting those exceptions into diagnostics, whereas the CLI would just allow the exception to halt execution. The CLI will also halt on error-level diagnostics, so this change allowed me to move more configuration management logic into Bicep.Core while not affecting user-visible behavior in either the lang server or the CLI (or at least only in ways I would consider cosmetic rather than substantive).